core/constraints/constraints.go

```diff
@@ -110,6 +110,12 @@ type Value struct {
 	// The default behaviour if the value is not specified is to allocate
 	// a public IP so that public cloud behaviour works out of the box.
 	AllocatePublicIP *bool `json:"allocate-public-ip,omitempty" yaml:"allocate-public-ip,omitempty"`
+
+	// AllocateIPv6, if nil or true, signals that a best effort should be made
+	// to allocate the machine with ipv6 networking where it can be achieved.
+	// The default behaviour if the value is not specified is to make a best
+	// effort approach to allocated with ipv6.
+	AllocateIPv6 *bool `json:"allocate-ipv6,omitempty" yaml:"allocate-ipv6,omitempty"`
```

Comment:
Is this actually a constraint or a behaviour. I'm not saying it's bad to add here, as we have `AllocatePublicIP` as a constraint, which doesn't feel like a constraint, more akin to a configuration. I wonder if these "constraints' are actually model-config?

Reply:
Good question. I don't know the answer but will ponder it today.

Reply:
@jameinel thoughts on this?

Reply:
Talked with Tom, constraint not the thing we want to do here. This is an internal ec2 implementation detail which should remain that way, rather than breaking encapsulation and surfacing to a public interface. Will be reworked.